lib/galaxy/tools/deps/resolvers/conda.py

```diff
@@ -104,21 +106,31 @@ def ensure_conda_installed(self):
         Make sure that conda is installed, and if conda can't be installed, mark resolver as disabled.
         We acquire a lock, so that multiple handlers do not attempt to install conda simultaneously.
         """
-        if not os.path.exists(self.dependency_manager.default_base_path):
-            os.mkdir(self.dependency_manager.default_base_path)
-        try:
-            with FileLock(os.path.join(self.dependency_manager.default_base_path, 'conda')):
-                if not self.conda_context.is_conda_installed():
-                    if self.auto_init:
-                        if self.conda_context.can_install_conda():
-                            if install_conda(self.conda_context):
-                                self.disabled = True
-                                log.warning("Conda installation requested and failed.")
-                        else:
+        target_path = self.conda_prefix_parent
+
+        def _check():
+            if not self.conda_context.is_conda_installed():
+                if self.auto_init:
+                    if self.conda_context.can_install_conda():
+                        if install_conda(self.conda_context):
                             self.disabled = True
+                            log.warning("Conda installation requested and failed.")
                     else:
                         self.disabled = True
-                        log.warning("Conda not installed and auto-installation disabled.")
+                else:
+                    self.disabled = True
+                    log.warning("Conda not installed and auto-installation disabled.")
+            else:
+                self.disabled = False
+
+        if not os.path.exists(target_path):
+            os.mkdir(target_path)
```

Comment:
Can this create concurrency problems if multiple handlers try to create the directory at the same time?